File: Assets/Scripts/Panel/PanelLogin/PanelLogin.cs
//================================
//===Author: easy
//===Email: gopiny@live.com
//===Date: 2017-09-25 10:23
//================================

using UnityEngine;
using Easy.FrameUnity.ESNetwork;
using Easy.FrameUnity.Util;

namespace Asobimo.Pachinko
{
    public class PanelLogin : MonoBehaviour
    {
        public UIInput Username;
        public UIInput Password;
        public UIButton BtnLogin;

        private RegisterDataReq _data;

        private void Awake()
        {
            this.RegisterBtnEvent();
            this.LoadLoginConfig();
            this.Init();
        }

        private void LoadLoginConfig()
        {
            _data = IOHelperUtil.ReadFromJson<RegisterDataReq>(URL.DEBUG_CONFIG);
            NetPackage.Sid = _data.Sid;
        }

        private void Init()
        {
            this.Username.value = _data.Username;
            this.Password.value = _data.Password;
        }

        public void RegisterBtnEvent()
        {
            EventDelegate.Add(this.BtnLogin.onClick, ()=>{
                    this.OnBtnLoginClick();
                    });
        }

        public void OnBtnLoginClick()
        {
            Net.Login(_data, Password.value, (res)=>{
                    Player.Inst.UserData = res.UserData;
                    //Player.Inst.BallsNum = res.UserData.BallsNum;
					NetPackage.Sid = res.SessionId;
                    _data.Sid = res.SessionId;
                    IOHelperUtil.SaveToJson<RegisterDataReq>(_data, URL.DEBUG_CONFIG);

					Net.Heartbeat();
                    ScenesManager.Inst.EnterLoadingScene(SceneName.E_SceneGame_1);
                    });
        }
    }
}

File: HotfixExport/Lua/PanelLogin.lua
--================================
--===Author: easy
--===Email: gopiny@live.com
--===Date: 2017-11-06 12:25
--================================

require("ExtendGlobal")
local CSNet = CS.Easy.FrameUnity.ESNetwork

PanelLogin = class()

function PanelLogin:ctor()

end

function PanelLogin:RegisterBtnEvent()
    local ok, btn = self.cs.HotfixUIDic:TryGetValue("BtnLogin")
    btn = btn:GetComponent("UIButton")
    CS.EventDelegate.Add(btn.onClick, self.onBtnLoginClick)
end

function PanelLogin:TransformGameObject(gameObject)
    print("===>TransformGameObject: " .. gameObject.name);
    self.gameObject = gameObject;
    self.cs = gameObject:GetComponent("PanelHotfix")
    self:RegisterBtnEvent()
end

function PanelLogin:onBtnLoginClick()
    print("Login")
    local callback = function(res)
        print("Login callback")
        CS.ScenesManager.Inst:EnterLoadingScene(CS.SceneName.E_SceneGame_1)
    end
    _G.Net:login("Test", "565656", callback)
end
